### PR TITLE
Fixes escape of \t and \r

### LINKED ISSUE #30: ParseNode Failure while processing string literal
Present on **github.com/microsoft/kiota-serialization-json-go**  `v0.5.1` & `v0.5.2`
Related to: https://github.com/microsoft/kiota-serialization-json-go/pull/26

call: `js.NewJsonParseNodeFactory().GetRootParseNode("application/json", bytes)`
where
### bytes
```
{"id":"AAMkADhlNjU5MzMzLTUyYzItNDJmYi04ZTgzLTkzYTE1ZDg0OTRjOQBGAAAAAAC9muvJZ4uvRYeQkyPxt40qBwCDUn-uvuceTb3Zb9F345Q1AAAAAAEMAACDUn-uvuceTb3Zb9F345Q1AAAuZPpRAAA=","@odata.context":"https://graph.microsoft.com/v1.0/$metadata#users('67fc108d-cc22-4a54-96af-7b91c5d34cf2')/messages/$entity","@odata.etag":"W/\"CQAAABYAAACDUn/uvuceTb3Zb9F345Q1AAAuT0ok\"","categories":[],"changeKey":"CQAAABYAAACDUn/uvuceTb3Zb9F345Q1AAAuT0ok","createdDateTime":"2022-06-15T05:27:05Z","lastModifiedDateTime":"2022-06-15T05:27:07Z","bccRecipients":[],"body":{"content":"<html lang=\"en\" style=\"min-height:100%; background:#ffffff\"><head>
<meta http-equiv=\"Content-Type\" content=\"text/html; charset=utf-8\"><meta name=\"viewport\" content=\"width=device-width\"><meta name=\"eventId\" content=\"aad-identity-protection-weekly-digest-report-v2\"><meta name=\"messageId\" content=\"d4db577e-fe10-4bea-8e6d-164bb1ebb039\"><meta name=\"cloud\" content=\"AZ\"><style id=\"mediaqueries\">
<!--
@media only screen and (max-width: 640px) {
.wrap-dangler
	{padding-right:10%!important}
.wrap-dangler.small-text-center .wrap-dangler.text-center
```
Receiving error: `invalid character '\r' in string literal `

### PR DESCRIPTION
**Pull request by dadams39**
Fixes #30 
- Did not add any tests. Passes all current tests. I used a file to upload a full file with \r and \t to ensure that a valid json was output.

### CODE REVIEW

File: json_serialization_writer.go [hunk 1]
```diff
@@ -27,6 +27,8 @@ func (w *JsonSerializationWriter) writeRawValue(value string) {
 	w.writer = append(w.writer, value)
 }
 func (w *JsonSerializationWriter) writeStringValue(value string) {
+	value = strings.ReplaceAll(strings.ReplaceAll(value, "\t", ""), "\r", "")
```
Comment (baywet) [inline json_serialization_writer.go:30]:
```suggestion
	value = strings.ReplaceAll(strings.ReplaceAll(value, "\t", "\\t"), "\r", "\\r")
```

(and move this below the next one so it doesn't end up escaping the \ multiple times)

**Review by baywet — CHANGES_REQUESTED** [review]
Thanks for the contribution.
Can you add test cases please?
Can you add an entry the changelog please? (you can take 0.5.5, and date it from today)

**Comment by dadams39** [comment]
Test Suite and Changelog updated.

**Review by baywet — APPROVED** [review]
Thanks a lot for the continuous conversation and the contribution!

**Pull request merged**